src/server/game/Spells/Spell.cpp

```diff
@@ -5380,7 +5380,15 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
             case SPELL_EFFECT_POWER_DRAIN:
             {
                 // Can be area effect, Check only for players and not check if target - caster (spell can have multiply drain/burn effects)
-                if (m_caster->GetTypeId() == TYPEID_PLAYER)
+                // effect target can be caster and can be different than spell target (e.g. Ferocious Bite)
```

Comment:
Don't you have a separate PR for this part?

Reply:
I do, but this PR depends on it.
That's wyh I added the "Dependant On:" part in the OP and linked the other issue.